libraries/common/effekt.effekt

```diff
@@ -557,6 +583,32 @@ def infixAnd { first: => Bool } { second: => Bool }: Bool =
   if (first()) second() else false
 
 
+// Bitwise operations
+// ===============
+extern pure def bitwiseShl(x: Int, y: Int): Int =
```

Comment:
I would have added `infixShiftLeft` and << to the parser, but that's fine for now. 

Trivia, did you know that apparently it was called `shl` in Pascal (written infix)?

Reply:
We will need more bitwise operations and I don't know a good design yet.

Reply:
Which ones are you thinking of?

Reply:
`bitwiseAnd`, `bitwiseOr`, `bitwiseNot`, `bitSet`, `bitGet`, sign extended shift right, zero extended shift right, literals, count leading zeroes, rank, select, ...

Reply:
We could at least add:

- `<<` shift left
- `>>` shift right
- `|` or
- `&` and
- `^` xor

and potentially
- `>>>` unsigned shift right (that's what JS does)

Reply:
We do not allow prefix operators at the moment so bitwise negation would need to be a function.